packages/store/src/appSlice/index.ts

```diff
@@ -90,32 +168,32 @@ export const appSlice = createSlice({
   name: sliceName,
   initialState,
   reducers: {
-    setFeatures: (state, action) => {
+    setFeatures: (state, action: PayloadAction<Record<string, boolean>>) => {
       state.features = {
         ...state.features,
         ...action.payload
       };
     },
-    setSnackbar: (state, { payload }) => {
-      let { message, autoHideDuration, action, children, onClick, onClose } = payload;
-      if (typeof payload === 'string' || payload instanceof String) {
-        message = payload;
-      }
+    setSnackbar: (state, { payload }: PayloadAction<SnackbarContent | string>) => {
+      const isString = typeof payload === 'string';
+
+      const message = isString ? payload : payload.message;
+      const autoHideDuration = isString ? undefined : payload.autoHideDuration;
+      const action = isString ? undefined : payload.action;
+      const preventClickToCopy = isString ? false : (payload.preventClickToCopy ?? false);
```

Comment:
```suggestion
      let nextSnackbar = {};
      if (typeof payload === 'string') {
        nextSnackbar = { action: undefined, autoHideDuration: undefined, message: payload.message,  preventClickToCopy: false };
      } else {
        nextSnackbar = payload;
      }
      state.snackbar = nextSnackbar
```
or to align with what we have:
```suggestion
      if (typeof payload === 'string') {
        state.snackbar.message = payload;
        return;
      }
      state.snackbar = payload;
```
maybe? just to get around the repetition...